exercises/list-ops/test/test_list_ops.c

```diff
@@ -311,11 +311,12 @@ static void test_reverse_empty_list(void)
    list_t *list = new_list(0, NULL);
    list_t *expected = new_list(0, NULL);
 
-   reverse_list(list);
-   check_lists_match(expected, list);
+   list_t *actual = reverse_list(list);
+   check_lists_match(expected, actual);
 
    delete_list(list);
    delete_list(expected);
+   delete_list(actual);
```

Comment:
Not sure what's more appropriate, but removing either `delete_list(list)` or `delete_list(actual)` causes the memcheck to pass with the current given `example.c`. This applies to the non-empty test as well.

Reply:
Can you help me understand what you mean here? It looks like all three of these deletions are necessary.

Reply:
The memcheck was originally failing with all three deletions with the original `example.c` but passed if you deleted one of them. I changed `remove_list()` in the example file to return a new `list_t` instead of mutating the one passed to the function. Everything passes fine now.